x-pack/plugins/apm/server/lib/anomaly_detection/create_anomaly_detection_jobs.ts

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Logger } from 'kibana/server';
+import uuid from 'uuid/v4';
+import { PromiseReturnType } from '../../../../observability/typings/common';
+import { Setup } from '../helpers/setup_request';
+import {
+  SERVICE_ENVIRONMENT,
+  TRANSACTION_DURATION,
+  PROCESSOR_EVENT,
+} from '../../../common/elasticsearch_fieldnames';
+import { ENVIRONMENT_NOT_DEFINED } from '../../../common/environment_filter_values';
+
+const ML_MODULE_ID_APM_TRANSACTION = 'apm_transaction';
+export const ML_GROUP_NAME_APM = 'apm';
+
+export type CreateAnomalyDetectionJobsAPIResponse = PromiseReturnType<
+  typeof createAnomalyDetectionJobs
+>;
+export async function createAnomalyDetectionJobs(
+  setup: Setup,
+  environments: string[],
+  logger: Logger
+) {
+  const { ml, config } = setup;
+  if (!ml) {
+    logger.warn('Anomaly detection plugin is not available.');
+    return [];
+  }
+  const mlCapabilities = await ml.mlSystem.mlCapabilities();
+  if (!mlCapabilities.mlFeatureEnabledInSpace) {
+    logger.warn('Anomaly detection feature is not enabled for the space.');
+    return [];
+  }
+  if (!mlCapabilities.isPlatinumOrTrialLicense) {
+    logger.warn(
+      'Unable to create anomaly detection jobs due to insufficient license.'
+    );
+    return [];
+  }
+  logger.info(
+    `Creating ML anomaly detection jobs for environments: [${environments}].`
+  );
+
+  const indexPatternName = config['apm_oss.transactionIndices']; // TODO [ML] - Do we want to use the config index name?
```

Comment:
@sqren what do u think about this? As it is, it will only create jobs where is the index pattern is determined by what's configured in kibana.yml. But we also have configurable UI indices in our settings page.

Reply:
This is already taken care of in [setup request](https://github.com/elastic/kibana/blob/2eb0896415122264fe23a76945d969ac74330b52/x-pack/plugins/apm/server/lib/helpers/setup_request.ts#L73-L79).


```ts
const { indices } = setup;
const indexPatternName = indicies['apm_oss.transactionIndices'];
```

Reply:
My concern is that the ML job datafeed would be using a different index depending on who created it. For example, If there are 2 different spaces each with it's own index pattern for transactions, you could create 2 different jobs with the same environment name but different indexPatternNames, so in the service maps or transaction duration chart, they will not know which of those 2 jobs is providing them with anomaly data.